src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -820,17 +811,16 @@ public void onQueryPhase(SearchContext searchContext, long tookInNanos) {
                         return;
                     }
 
-                    @SuppressWarnings("unchecked")
-                    final Map<String, Set<String>> maskedFieldsMap = (Map<String, Set<String>>) HeaderHelper.deserializeSafeFromHeader(
-                        threadPool.getThreadContext(),
-                        ConfigConstants.OPENDISTRO_SECURITY_MASKED_FIELD_HEADER
-                    );
-                    final String maskedEval = SecurityUtils.evalMap(maskedFieldsMap, indexModule.getIndex().getName());
-                    if (maskedEval != null) {
-                        final Set<String> mf = maskedFieldsMap.get(maskedEval);
-                        if (mf != null && !mf.isEmpty()) {
+                    try {
+                        if (dlsFlsValve.hasFieldMasking(indexModule.getIndex().getName())) {
                             dlsFlsValve.onQueryPhase(queryResult);
                         }
+                    } catch (PrivilegesEvaluationException e) {
+                        log.error("Error while evaluating field masking config", e);
+                        // It is safe to call the code nevertheless, as this code does not enforce any privileges.
+                        // Rather, it performs some fixes to keep aggregations happy after field masking has been
+                        // applied. If no field masking has been applied, this should be a no-op.
+                        dlsFlsValve.onQueryPhase(queryResult);
```

Comment:
To be honest, I believe that in a real world scenario, it is not possible to hit this code. Any exception that would be thrown here would also been thrown in some stage before the actual request processing - which would have request processing aborted all together. 

However, this is hard to prove. There might be weird special cases, maybe race-conditions, which could lead to this exception been thrown here. That's why we need to handle it at least in some way. I believe the way it is currently handled, is fine, just as described in the comment.

Thus, IMHO, creating a test case for this is not really woth the effort - at least when considering the other potential areas of work.

Reply:
Of course, one could create a unit-test with the right conditions being mocked together. However, I think it would require a bit of refactoring to make this code unit-testable.

Reply:
Marking this thread as resolved.